File: OpenDentalWeb/OpenDental/src/com/opendental/odweb/client/remoting/SerializeStringEscapes.java
package com.opendental.odweb.client.remoting;

public class SerializeStringEscapes {
	
	/** Escapes common characters used in XML from the passed in String. */
	public static String EscapeForXml(String myString) {
		StringBuilder strBuild=new StringBuilder();
		int length=myString.length();
		for(int i=0;i<length;i++) {
			String character=myString.substring(i,i+1);
			if(character.equals("<")) { 
				strBuild.append("&lt;"); 
				continue;
			}
			else if(character.equals(">")) { 
				strBuild.append("&gt;"); 
				continue; 
			}
			else if(character.equals("\"")) { 
				strBuild.append("&quot;"); 
				continue; 
			}
			else if(character.equals("\'")) { 
				strBuild.append("&#039;"); 
				continue;
			}
			else if(character.equals("&")) { 
				strBuild.append("&amp;"); 
				continue; 
			}
			strBuild.append(character);
		}
    return strBuild.toString();
	}
	
	
}

File: OpenDentalWebService/Remoting/SerializeStringEscapes.cs
using System;
using System.Text;

namespace OpenDentalWebService {
	public class SerializeStringEscapes {

		///<summary>Escapes common characters used in XML from the passed in String.</summary>
		public static string EscapeForXml(string myString) {
			StringBuilder strBuild=new StringBuilder();
			int length=myString.Length;
			for(int i=0;i<length;i++) {
				String character=myString.Substring(i,1);
				if(character.Equals("<")) {
					strBuild.Append("&lt;");
					continue;
				}
				else if(character.Equals(">")) {
					strBuild.Append("&gt;");
					continue;
				}
				else if(character.Equals("\"")) {
					strBuild.Append("&quot;");
					continue;
				}
				else if(character.Equals("\'")) {
					strBuild.Append("&#039;");
					continue;
				}
				else if(character.Equals("&")) {
					strBuild.Append("&amp;");
					continue;
				}
				strBuild.Append(character);
			}
			return strBuild.ToString();
		}


	}
}